chaosaws/asg/probes.py

```diff
@@ -59,6 +63,179 @@ def desired_equals_healthy_tags(tags: List[Dict[str, str]],
             "Non-empty tags is required")
 
     client = aws_client('autoscaling', configuration, secrets)
+    groups_descr = get_asg_by_tags(tags, client)
+
+    return is_desired_equals_healthy(groups_descr)
+
+
+def wait_desired_equals_healthy(asg_names: List[str],
+                                configuration: Configuration = None,
+                                timeout: int = 300,
+                                secrets: Secrets = None) -> AWSResponse:
+    """
+    Wait until desired number matches the number of healthy instances
+
+    for each of the auto-scaling groups
+
+    Returns: Boolean, Integer (number of seconds it took to wait)
+    """
+
+    if not asg_names:
+        raise FailedActivity(
+            "Non-empty list of auto scaling groups is required")
+
+    client = aws_client('autoscaling', configuration, secrets)
+
+    start = time.time()
+
+    while True:
+        groups_descr = client.describe_auto_scaling_groups(
+            AutoScalingGroupNames=asg_names)
+        result = is_desired_equals_healthy(groups_descr)
+
+        if (time.time() - start) > timeout:
+            logger.info("Timed out")
+
+            return maxsize
+
+        if result:
+            waiting_time = int(time.time() - start)
+            logger.info("Waiting time was: {}".format(waiting_time))
+
+            return waiting_time
+        time.sleep(0.1)
+
+
+def wait_desired_not_equals_healthy_tags(tags: List[Dict[str, str]],
+                                         timeout: int = 300,
+                                         configuration: Configuration = None,
+                                         secrets: Secrets = None
+                                         ) -> AWSResponse:
+    """
+    Wait until desired number doesn't match the number of healthy instances
+
+    for each of the auto-scaling groups matching tags provided
+
+    `tags` are  expected as:
+    [{
+        'Key': 'KeyName',
+        'Value': 'KeyValue'
+    },
+    ...
+    ]
+
+    Returns: Integer (number of seconds it took to wait)
+    or sys.maxsize in case of timeout
+    """
+
+    if not tags:
+        raise FailedActivity(
+            "Non-empty tags is required")
+
+    client = aws_client('autoscaling', configuration, secrets)
+
+    start = time.time()
+
+    while True:
+        groups_descr = get_asg_by_tags(tags, client)
+        logger.debug(groups_descr)
+        result = is_desired_equals_healthy(groups_descr)
+
+        if (time.time() - start) > timeout:
+            logger.info("Timed out")
+
+            return maxsize
+
+        if not result:
+            waiting_time = int(time.time() - start)
+            logger.info("Waiting time was: {}".format(waiting_time))
```

Comment:
I'm sorry I'll have to bother you again. The info level is not good here because that is printed to the console as well as part of the toolkit's console log. This is more of a debug message here.

Do you mind turning this to debug?

Also, while I appreciate the message, I'm more interested if the log tells me what this means when it can. The rationale is that it helps investigating.

Again sorry to bother you here.